keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlResourceRepository.kt

```diff
@@ -121,35 +125,36 @@ open class SqlResourceRepository(
   }
 
   // todo: this is not retryable due to overall repository structure: https://github.com/spinnaker/keel/issues/740
-  override fun store(resource: Resource<*>) {
-    val uid = jooq.select(RESOURCE.UID)
+  override fun store(resource: Resource<*>): Resource<*> {
+    val uid = randomUID().toString()
+    val version = jooq
+      .select(
+        coalesce(
+          max(RESOURCE.VERSION).plus(1),
+          value(1)
+        )
+      )
```

Comment:
When updating we want current version + 1, when creating for the first time we want 1

Reply:
What's the purpose of `max()` in the expression?

Reply:
SQL's `max` function will return the highest value of that field from all the selected rows. So in this case the highest version number for any resource with the same `id`.